js/ui/file_uploader.js

```diff
@@ -1119,16 +1125,125 @@ var FileUploader = Editor.inherit({
+        return result;
+    },
+    _sendFileChunk: function(file, chunksData) {
+        var that = this;
+        var chunk = chunksData.chunks.shift();
+        if(chunk) {
+            chunksData.loadedBytes += chunk.blob.size;
+
+            ajax.sendRequest({
+                url: this.option("uploadUrl"),
+                method: this.option("uploadMethod"),
+                headers: this.option("uploadHeaders"),
+                beforeSend: function(xhr) {
+                    file.request = xhr;
+                },
+                upload: {
+                    "onloadstart": function() {
+                        if(!file.isStartLoad) {
+                            file.isStartLoad = true;
+                            file.onLoadStart.fire();
+                        }
+                    },
+                    "onabort": function() {
+                        file.onAbort.fire();
+                        chunksData.chunks = [];
+                    }
+                },
+                data: this._createChunkFormData(chunksData.name, this.option("name"),
+                    chunk.blob,
+                    chunk.index,
+                    chunksData.count,
+                    chunksData.type,
+                    chunksData.guid,
+                    chunksData.fileSize)
+            }).done(function() {
+                file.onProgress.fire({
+                    loaded: chunksData.loadedBytes,
+                    total: file.value.size
+                });
+                if(chunksData.chunks.length === 0) {
+                    file.onLoad.fire();
+                }
+                that._sendFileChunk(file, chunksData);
+            }).fail(function(e) {
+                if(that._isStatusError(e.status)) {
+                    file._isError = true;
+                    file.onError.fire();
+                }
+                chunksData.chunks = [];
+            });
+        }
+    },
+    _createChunkFormData: function(fileName, blobName, blob, index, count, type, guid, size) {
+        var formData = new window.FormData();
+        formData.append(blobName, blob);
+        formData.append("metaData", JSON.stringify({
+            Name: fileName,
+            Index: index,
+            Count: count,
+            FileSize: size,
+            Type: type,
+            Guid: guid
+        }));
+        return formData;
+    },
 
-        var $file = file.$file,
-            value = file.value;
-
-        if($file) {
-            file.progressBar = this._createProgressBar(value.size);
-            file.progressBar.$element().appendTo($file);
+    _sliceFile: function(file, startPos, length) {
+        if(file.slice) {
+            return file.slice(startPos, startPos + length);
+        }
+        if(file.webkitSlice) {
+            return file.webkitSlice(startPos, startPos + length);
+        }
+        if(file.mozSlice) {
+            return file.mozSlice(startPos, startPos + length);
+        }
+        return null;
```

Comment:
Maybe to make easier:
```
var slice = file.size || file.webkitSlice || file.mozSlice;
return typeUtils.isDefined(slice) ? slice(startPos, startPos + length) : null;

```

Reply:
This approach will not work.
In this case, it is necessary to write: `slice.call(file, startPos, startPos + length)` - looks not pretty